### DISCUSSION TITLE
execute job ONLY on foreground

### DISCUSSION THREAD

**Discussion by bodeg** [shinyorg/shiny#694] [Q&A] [answered]
How can I set the job to execute ONLY when the app is running on the foreground?

```
// I set 
JobInfo.RunOnForeground = true
// and call
services.UseJobForegroundService()
```

The job executes on foreground, but it still running in the background.

Thanks. alon

**Comment by aritchie** [comment, marked as answer]
This doesn't exist.  You would have to do these checks internal to your job.
